app/src/main/java/com/asfoundation/wallet/repository/InAppPurchaseService.java

```diff
@@ -178,10 +228,16 @@ public void start() {
                 .toSingleDefault(paymentTransaction))
                 .filter(transaction -> transaction.getState()
                     .equals(PaymentTransaction.PaymentState.APPROVED))
-                .flatMapCompletable(transaction -> approveService.remove(transaction.getUri())
-                    .andThen(buyService.buy(transaction.getUri(), transaction)
-                        .onErrorResumeNext(throwable -> cache.save(transaction.getUri(),
-                            new PaymentTransaction(transaction, errorMapper.map(throwable))))))))
+                .flatMapCompletable(transaction -> {
+                  return transaction.getTransactionBuilder()
```

Comment:
Can be turned into a lambda expression OR extract the transaction.getUri(), to avoid calling it 4 times